### PR TITLE
HPCC-24795 ECL Watch open in new window URL issue

### PR DESCRIPTION
**Pull request by GordonSmith**
Rollback change from:  https://github.com/hpcc-systems/HPCC-Platform/commit/f468a00181569c6ef2d5a35e67702d289251f859

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq - this rolls back part of your PR:  https://github.com/hpcc-systems/HPCC-Platform/commit/f468a00181569c6ef2d5a35e67702d289251f859

Can you check it doesn't break anything?

**Comment by miguelvazq** [comment]
@GordonSmith This will break the recent filters open button that shows the last 5 filters the user has used. 

- Make sure environment has WsStore
- Go to WUQuery filter
- If no recent filters appear - filter by something to populate the recent filters list (right above form)
- Click on the open button which should pass in the what that filter was for that search

Currently, with this rollback we just open a new page with no filter added. Perhaps, a way to introduce routing here may help?

File: esp/src/eclwatch/_Widget.js [hunk 1]
```diff
@@ -130,22 +130,22 @@ define([
...
                 this.params.__filter = ioQuery.objectToQuery(filterParams);
             }
             var paramsString = ioQuery.objectToQuery(this.params);
             return baseUrl + "?" + paramsString;
         },
 
-        _onNewPage: function (filterFromComponent) {
-            var win = window.open(this.getURL(filterFromComponent), "_blank");
+        _onNewPage: function (event) {
+            var win = window.open(this.getURL(), "_blank");
```
Comment (miguelvazq) [inline esp/src/eclwatch/_Widget.js:148]: `let win;
            `filterFromComponent.currentTarget ?  win = window.open(this.getURL(), "_blank") : win = window.open(this.getURL(filterFromComponent), "_blank") ;`

File: esp/src/eclwatch/_Widget.js [hunk 2]
```diff
@@ -130,22 +130,22 @@ define([
...
 
-        getURL: function (filterFromComponent) {
+        getURL: function () {
             var baseUrl = document.URL.split("#")[0];
             var baseUrlParts = baseUrl.split("?");
             baseUrl = baseUrlParts[0];
             var args = baseUrlParts[1];
             delete this.params.__filter;
-            var filterParams = filterFromComponent ? filterFromComponent : this.getFilterParams();
+            var filterParams = this.getFilterParams();
```
Comment (miguelvazq) [inline esp/src/eclwatch/_Widget.js:139]: `var filterParams = !filterFromComponent ? this.getFilterParams() : filterFromComponent;` ?

File: esp/src/eclwatch/_Widget.js [hunk 3]
```diff
@@ -130,22 +130,22 @@ define([
             return retVal;
         },
 
-        getURL: function (filterFromComponent) {
+        getURL: function () {
```
Comment (miguelvazq) [inline esp/src/eclwatch/_Widget.js:133]: `getURL: function (filterFromComponent) {`

**Comment by miguelvazq** [comment]
I will accept this code review to fix the bug. I will open a new JIRA to allow the recent filters once that JIRA has been resolved.

**Review by miguelvazq — APPROVED** [review]

**Pull request merged**